channels_fields/serializers.py

```diff
@@ -1,40 +1,271 @@
+from channels_fields.models import FieldChannel, FieldList, Subfield
+from course_catalog.constants import PrivacyLevel
+from course_catalog.models import UserList
+from course_catalog.serializers import UserListSerializer
 from open_discussions.serializers import WriteableSerializerMethodField
 from profiles.models import Profile
 from widgets.models import WidgetList
 
 User = get_user_model()
 
+log = logging.getLogger(__name__)
+
+
+class FieldListSerializer(serializers.ModelSerializer):
+    """Serializer for FieldList"""
+
+    field_list = UserListSerializer(many=False, read_only=True)
+
+    class Meta:
+        model = FieldList
+        fields = "__all__"
+        read_only_fields = ("field_channel",)
+
+
+class SubfieldSerializer(serializers.ModelSerializer):
+    """Serializer for Subfields"""
+
+    parent_field = serializers.SlugRelatedField(
+        many=False, read_only=True, slug_field="name"
+    )
 
-class FieldChannelSerializer(serializers.ModelSerializer):
+    field_channel = serializers.SlugRelatedField(
+        many=False, read_only=True, slug_field="name"
+    )
+
+    class Meta:
+        model = Subfield
+        fields = ("parent_field", "field_channel", "position")
+
+
+class FieldChannelSerializer(ChannelAppearanceMixin, serializers.ModelSerializer):
     """Serializer for FieldChannel"""
 
+    lists = FieldListSerializer(many=True, read_only=True)
+    featured_list = UserListSerializer(many=False, read_only=True)
+    subfields = SubfieldSerializer(many=True, read_only=True)
+
     class Meta:
         model = FieldChannel
-        fields = "__all__"
-        read_only_fields = ("widget_list",)
+        fields = (
+            "name",
+            "title",
+            "about",
+            "public_description",
+            "subfields",
+            "featured_list",
+            "lists",
+            "about",
+            "avatar",
+            "avatar_medium",
+            "avatar_small",
+            "banner",
+            "widget_list",
+            "updated_on",
+            "created_on",
+            "id",
+            "ga_tracking_id",
+        )
+        read_only_fields = fields
+
+
+class FieldChannelCreateSerializer(serializers.ModelSerializer):
+    """Write serializer for FieldChannel"""
+
+    featured_list = serializers.PrimaryKeyRelatedField(
+        many=False,
+        allow_null=True,
+        allow_empty=True,
+        required=False,
+        queryset=UserList.objects.filter(privacy_level=PrivacyLevel.public.value),
+    )
+    lists = WriteableSerializerMethodField()
+    subfields = WriteableSerializerMethodField()
+
+    class Meta:
+        model = FieldChannel
+        fields = (
+            "name",
+            "title",
+            "public_description",
+            "subfields",
+            "featured_list",
+            "lists",
+            "about",
+        )
+
+    def validate_lists(self, lists):
+        """Validator for lists"""
```

Comment:
This is allowing `lists` to either be `number[]` or `FieldList[]`, right? 

**Generic Question:** Is this a common pattern in our APIs, that they accept either an ID value or an object value? It doesn't necessarily seem worth the extra complexity to me.

**Suggestion:** Maybe add more of a docstring, or a type annotation like `List[Union[number, FieldList]]`. _Not sure I have that python typing syntax quite right._

Reply:
I will change the code to expect a list of integers only instead of converting something else to that, it's what the frontend should send it once it gets written.  Ditto for subfields (list of strings ie field names).